test/boost/cdc_test.cc

```diff
@@ -427,27 +427,32 @@ SEASTAR_THREAD_TEST_CASE(test_primary_key_logging) {
     }, mk_cdc_test_config()).get();
 }
 
-SEASTAR_THREAD_TEST_CASE(test_pre_image_logging) {
+SEASTAR_THREAD_TEST_CASE(test_pre_post_image_logging) {
     do_with_cql_env_thread([](cql_test_env& e) {
         auto test = [&e] (bool enabled, bool with_ttl) {
-            cquery_nofail(e, "CREATE TABLE ks.tbl (pk int, pk2 int, ck int, ck2 int, val int, PRIMARY KEY((pk, pk2), ck, ck2)) WITH cdc = {'enabled':'true', 'preimage':'"s + (enabled ? "true" : "false") + "'}");
-            cquery_nofail(e, "UPDATE ks.tbl"s + (with_ttl ? " USING TTL 654" : "") + " SET val = 11111 WHERE pk=1 AND pk2=11 AND ck=111 AND ck2=1111");
+            cquery_nofail(e, format("CREATE TABLE ks.tbl (pk int, pk2 int, ck int, ck2 int, val int, val2 int, PRIMARY KEY((pk, pk2), ck, ck2)) "
+                "WITH cdc = {{'enabled':'true', 'preimage':'{0}', 'postimage':'{0}'}}", enabled));
+            cquery_nofail(e, "UPDATE ks.tbl"s + (with_ttl ? " USING TTL 654" : "") + " SET val = 11111, val2 = 22222 WHERE pk=1 AND pk2=11 AND ck=111 AND ck2=1111");
 
             auto rows = select_log(e, "tbl");
 
             BOOST_REQUIRE(to_bytes_filtered(*rows, cdc::operation::pre_image).empty());
+            BOOST_REQUIRE(to_bytes_filtered(*rows, cdc::operation::post_image).empty() == !enabled);
 
             auto first = to_bytes_filtered(*rows, cdc::operation::update);
 
             auto ck2_index = column_index(*rows, cdc::log_data_column_name("ck2"));
             auto val_index = column_index(*rows, cdc::log_data_column_name("val"));
+            auto val2_index = column_index(*rows, cdc::log_data_column_name("val2"));
```

Comment:
There's something I don't understand here. If we have both a post-image and a delta, like I think we have here, what does this search find, the post-image or the delta? Should the test search both, and verify that they are both correct? The code below seems to do exactly this, but this code doesn't.

Reply:
The code above just figures out column index in result sets for our cdc log queries. This is idempotent across all queries in a single test run, but might vary across builds/versions.